corehq/apps/settings/forms.py

```diff
@@ -346,7 +346,7 @@ def clean_expiration_date(self):
         if not self.cleaned_data['expiration_date']:
             return self.cleaned_data['expiration_date']
 
-        expiration_date = self.timezone.localize(self.cleaned_data['expiration_date'])
+        expiration_date = self.cleaned_data['expiration_date'].replace(tzinfo=self.timezone)
```

Comment:
Does this do the same thing that `timezone.localize(...)` did? I think you might want to use [`datetime.astimezone`](https://docs.python.org/3/library/datetime.html#datetime.datetime.astimezone).

Reply:
Yes, localize worked the same way -- it took a naive time and just tacked the given timezone to it. `replace` does the same thing, whereas `astimezone` assumes that naive object is UTC. I tested this locally, but when I'm using this widget, and I send across a date, I am sending that date across in my local time, and the time attached to it is '00:00:00' -- it is important that that does not first get adjusted to UTC

Reply:
It's probably also worth noting that this specific case is covered in _test_forms:HQApiKeyTests.test_expiration_date_is_localized_